strax/xenon/cut_plugins.py

```diff
@@ -0,0 +1,48 @@
+import numpy as np
+
+import strax
+export, __all__ = strax.exporter()
+
+
+class FiducialCylinder1T(strax.Plugin):
+    """Implementation of fiducial volume cylinder 1T,
+	ported from lax.sciencerun0.py"""
+    depends_on = ('event_positions',)
+    provides = 'fiducial_cylinder_1t'
+    dtype = [('cut_fiducial_cylinder', np.bool, 'One tonne fiducial cylinder')]
+
+    def compute(self, events):
+        arr = np.all([(-92.9 < events['z']), (-9 > events['z']),
+            (36.94 > np.sqrt(events['x']**2 + events['y']**2))], axis=0)
+        return dict(cut_fiducial_cylinder=arr)
+
+
+class S1MaxPMT(strax.LoopPlugin):
+    """Removes events where the largest hit in S1 is too large
+	port from lax.sciencerun0.py"""
+    depends_on = ('events', 'event_basics', 'peak_basics')
+    dtype = [('cut_s1_max_pmt', np.bool, 'S1 max PMT cut')]
+
+    def compute_loop(self, event, peaks):
+        ret = dict(cut_s1_max_pmt=True)
+        if not len(peaks) or np.isnan(event['s1_index']):
+            return ret
+
+        peak = peaks[event['s1_index']]
+        max_channel = peak['max_pmt_area']
+        ret['cut_s1_max_pmt']=(max_channel < 0.052 * event['s1_area'] + 4.15)
+        return ret
+
+
+class S1LowEnergyRange(strax.Plugin):
+    """Pass only events with cs1<200"""
+    depends_on = ('events', 'corrected_areas')
+    dtype = [('cut_s1_low_energy_range', np.bool, "Event under 200pe")]
+
+    def compute(self, events):
+         ret = np.all([events['cs1'] < 200], axis=0)
```

Comment:
Why do you need np.all here? I'm not sure what happens now, sounds like you're computing one boolean per chunk. I would have thought you'd just do `ret = events['cs1'] < 200`

Reply:
Ah, I see, the extra []'s make a new axis, and you're all-ing over that axis which has one element here (so it's not needed here), but above you use it as a way of combining event selections. 

OK, I usually use the & sign for this, e.g. (cs1 > 70) & (r < 20) % ... .